### PR TITLE
update redux and redux toolkit

### LINKED ISSUE #3287: Update Redux/ReduxToolkit
**Describe the task**
Major updates to redux and redux toolkit depends on running react 18.

**Acceptance Criteria**
- [x] Products run with the redux v9 and redux toolkit v2

**Additional context**
- Redux: https://github.com/bcgov/wps/pull/3284
- Redux Toolkit: https://github.com/bcgov/wps/pull/3283

**Blocked By**
- https://github.com/bcgov/wps/issues/1845

### PR DESCRIPTION
**Pull request by jmetev1**
closes #3287
This updates redux and redux toolkit and makes a few small changes required for that update.

# Test Links:
[Landing Page](https://wps-pr-3976-e1e498-dev.apps.silver.devops.gov.bc.ca/)
[MoreCast](https://wps-pr-3976-e1e498-dev.apps.silver.devops.gov.bc.ca/morecast)
[Percentile Calculator](https://wps-pr-3976-e1e498-dev.apps.silver.devops.gov.bc.ca/percentile-calculator)
[C-Haines](https://wps-pr-3976-e1e498-dev.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-3976-e1e498-dev.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-3976-e1e498-dev.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN,s=9999&f=c7&c=NaN&w=NaN)
[Auto Spatial Advisory (ASA)](https://wps-pr-3976-e1e498-dev.apps.silver.devops.gov.bc.ca/auto-spatial-advisory)
[HFI Calculator](https://wps-pr-3976-e1e498-dev.apps.silver.devops.gov.bc.ca/hfi-calculator)

### CODE REVIEW

File: web/src/features/moreCast2/components/editInputCell.test.tsx [hunk 1]
```diff
@@ -77,45 +77,50 @@ describe('EditInputCell', () => {
...
       <GridApiContext.Provider value={apiMock}>
         <EditInputCell {...defaultProps} id={1} value="10" field="test" hasFocus={false} error="" />
       </GridApiContext.Provider>
     )
 
     const input = within(getByTestId('forecast-edit-cell')).getByRole('spinbutton') as HTMLInputElement
-    input.focus()
-    fireEvent.blur(input)
-
+    await act(async () => {
```
Comment (jmetev1) [inline web/src/features/moreCast2/components/editInputCell.test.tsx:88]: this gets rid of the warnings ```Warning: An update to ForwardRef(FormControl) inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
  /* fire events that update state */
});
/* assert on the output */```

File: web/src/features/moreCast2/slices/dataSlice.test.ts [hunk 2]
```diff
@@ -89,7 +89,7 @@ describe('dataSlice', () => {
   describe('reducer', () => {
     const dummyError = 'an error'
     it('should be initialized with correct state flags', () => {
-      expect(dataSliceReducer(undefined, { type: undefined })).toEqual(initialState)
+      expect(dataSliceReducer(undefined, { type: '' })).toEqual(initialState)
```
Comment (jmetev1) [inline web/src/features/moreCast2/slices/dataSlice.test.ts:92]: redux updates mean type has to be a string now

File: web/src/features/auth/slices/authenticationSlice.test.ts [hunk 3]
```diff
@@ -53,7 +53,7 @@ describe('authenticationSlice', () => {
     it('should be initialized with correct state', () => {
       expect(
         authReducer(undefined, {
-          type: undefined
+          type: ''
```
Comment (jmetev1) [inline web/src/features/auth/slices/authenticationSlice.test.ts:56]: with new redux this has to be a string

File: web/src/features/moreCast2/components/editInputCell.test.tsx [hunk 4]
```diff
@@ -44,8 +44,8 @@ vi.mock('@/features/moreCast2/slices/validInputSlice', () => ({
 const mockDispatch = vi.fn()
 
 beforeEach(() => {
-  vi.clearAllMocks()
-  ;(useDispatch as jest.Mock).mockReturnValue(mockDispatch)
+  vi.clearAllMocks();
+  (useDispatch as unknown as jest.Mock).mockReturnValue(mockDispatch)
```
Comment (jmetev1) [inline web/src/features/moreCast2/components/editInputCell.test.tsx:48]: something changed about the typing of usedispatch. is this ok?

File: web/tsconfig.app.json [hunk 5]
```diff
@@ -36,36 +29,16 @@
...
-      ],
-    },
+      "@/*": ["./src/*"],
+      "#root/*": ["./*"],
+      "app/*": ["./src/app/*"],
+      "utils/*": ["./src/utils/*"],
+      "api/*": ["./src/api/*"],
+      "features/*": ["./src/features/*"],
+      "components/*": ["src/components/*"],
+      "components": ["./src/components/index"],
```
Comment (jmetev1) [inline web/tsconfig.app.json:39]: prettier did this. i could undo it suppose. there's a npm format command but i don't think it's part of ci or anything. is that on purpose?

Comment (conbrad) [reply web/tsconfig.app.json:39]: Yeah we'll have to align that in our lint check perhaps, but we don't want any code changes during CI.

**Review by conbrad — APPROVED** [review]
Nice work, thank you!

**Review by brettedw — APPROVED** [review]
Thanks @jmetev1!

**Review by dgboss — APPROVED** [review]
Looks good! Thank you!

**Pull request merged**